Gems/Atom/Feature/Common/Code/Source/Material/SubsurfaceTransmissionParameterFunctor.cpp

```diff
@@ -83,9 +83,44 @@ namespace AZ
             transmissionTintThickness.Set(tintColor.GetAsVector3());
             transmissionTintThickness.SetW(thickness);
 
-            context.GetShaderResourceGroup()->SetConstant(m_scatterDistance, scatterDistance);
-            context.GetShaderResourceGroup()->SetConstant(m_transmissionParams, transmissionParams);
-            context.GetShaderResourceGroup()->SetConstant(m_transmissionTintThickness, transmissionTintThickness);
+            context.GetMaterialShaderParameter()->SetParameter(m_scatterDistance, scatterDistance);
+            context.GetMaterialShaderParameter()->SetParameter(m_transmissionParams, transmissionParams);
+            context.GetMaterialShaderParameter()->SetParameter(m_transmissionTintThickness, transmissionTintThickness);
+        }
+
+        bool SubsurfaceTransmissionParameterFunctor::UpdateShaderParameterConnections(const RPI::MaterialShaderParameterLayout* layout)
+        {
+            bool valid = true;
+            if (m_scatterDistance.ValidateOrFindIndex(layout) == false)
+            {
+                AZ_Error(
+                    "SubsurfaceTransmissionParameterFunctorSourceData",
+                    false,
+                    "Could not find shader parameter '%s'",
+                    m_scatterDistance.GetName().GetCStr());
+                valid &= false;
+            }
+
+            if (m_transmissionParams.ValidateOrFindIndex(layout) == false)
+            {
+                AZ_Error(
+                    "SubsurfaceTransmissionParameterFunctorSourceData",
+                    false,
+                    "Could not find shader parameter '%s'",
+                    m_transmissionParams.GetName().GetCStr());
+                valid &= false;
+            }
+
+            if (m_transmissionTintThickness.ValidateOrFindIndex(layout) == false)
+            {
+                AZ_Error(
+                    "SubsurfaceTransmissionParameterFunctorSourceData",
+                    false,
+                    "Could not find shader parameter '%s'",
+                    m_transmissionTintThickness.GetName().GetCStr());
+                valid &= false;
+            }
+            return valid;
```

Comment:
Is the goal here to try all and report all errors (as implemented) or would it be better to fail fast (return false immediately)? Saw this pattern multiple times.

Reply:
The goal is to report all errors, since changes that fix these errors usually cause a rebuild of many (if not all) material-shaders, which can take several minutes.
The ability to see and fix several very similar errors all at once quickly became very desirable during these refactorings.